apps/gitness/src/framework/context/AppContext.tsx

```diff
@@ -36,7 +36,10 @@ export const AppProvider: React.FC<{ children: ReactNode }> = ({ children }) =>
         switch (response.status) {
           case 401:
             localStorage.removeItem('token')
-            window.location.href = '/signin'
+            //prevent infinite loop for redirect signin page
+            if (window.location.pathname !== '/signin') {
```

Comment:
while this works, this is not the correct solution. we should find the root cause and fix that.

Reply:
Added a fix here : https://github.com/harness/canary/pull/286, closing this PR.